modules/function/src/main/java/com/opengamma/strata/function/calculation/swaption/SwaptionCalculationFunction.java

```diff
@@ -90,17 +90,22 @@ public FunctionRequirements requirements(
       CalculationParameters parameters,
       ReferenceData refData) {
 
+    // extract data from product
     Swaption product = trade.getProduct();
-
     IborIndex index = product.getIndex();
-    DiscountCurveKey dfKey = DiscountCurveKey.of(product.getCurrency());
-    IborIndexCurveKey iborKey = IborIndexCurveKey.of(index);
-    SwaptionVolatilitiesKey volKey = SwaptionVolatilitiesKey.of(index);
-    IndexRateKey iborRateKey = IndexRateKey.of(index);
 
+    // use lookup to build requirements
+    SwaptionMarketLookup swaptionLookup = parameters.getParameter(SwaptionMarketLookup.class);
+    FunctionRequirements swaptionReqs = swaptionLookup.requirements(ImmutableSet.of(index));
```

Comment:
This makes me thinks that `SwaptionMarketLookup.requirements` should have a varargs parameter.